### DISCUSSION TITLE
Setting second lift and retract distances and speeds for Photon Workshop files doesn't seem to work

### DISCUSSION THREAD

**Discussion by kozross** [sn4k3/UVtools#934] [Q&A] [answered]
If I use the specified Printer Notes keywords to try and set these parameters (for example, putting `LiftHeight2_3` into the Notes section), these do not get set when the resulting `.sl1` is converted. However, I know that this format supports such settings (Chitubox and Lychee both set these just fine), and indeed, I can change them manually if I open the converted file in UVTools' GUI.

Am I missing something or is this a missing feature?

**Comment by sn4k3** [comment]
Send me the SL1 so I can debug the problem.

**Reply by kozross** [reply]
Here is a [simple project](https://github.com/user-attachments/files/17198093/project_0.05mm_48m.zip): this produces the following [`.sl1`](https://github.com/user-attachments/files/17198095/project_sl1.zip), and [`.pm3n`](https://github.com/user-attachments/files/17198099/project_pm3n.zip) after running `uvtools --cmd convert` with `PhotonWorkshopFile` as the requested format. 

Notably, this is what I had in the Printer Notes:

![Screenshot From 2024-10-01 14-15-32](https://github.com/user-attachments/assets/ac39f5a5-145b-4930-8f88-db3bd3d0dd71)

Namely, first lift height is 3mm, second lift height is also 3mm, first lift speed is 180mm/min (3mm/s), and second lift speed is 240mm/min (5mm/s). If I examine the result in UVTools' GUI, the first lift speed and height are indeed correct:

![Screenshot From 2024-10-01 14-16-43](https://github.com/user-attachments/assets/d2cc628a-7031-4844-8f03-04a05335846e)

But the second lift height is set to 0 (but the _speed_ is correct):

![Screenshot From 2024-10-01 14-17-06](https://github.com/user-attachments/assets/7622ee53-84b4-455a-85ca-3416de4cb545)

I've noticed the same if I set a different bottom layer retract speed (first or second).

**Reply by sn4k3** [reply, marked as answer]
At conversion time it comes without version (defaults to 1) which does not support TMSC and set values in other way. It then before encoding sanatize version to 517, but thats too late since checks was made before.

To solve this add: `FILEVERSION_517` after `FILEFORMAT_PM3N`.

On next release I will introduce a modification to sanitize file version before convertion and then will allow to omit FILEVERSION for your and other anycubic machines.

**Reply by kozross** [reply]
Fantastic, thank you!
